src/python/pants/backend/javascript/dependency_inference/rules.py

```diff
@@ -174,8 +173,14 @@ def _add_extensions(file_imports: frozenset[str]) -> PathGlobs:
 async def _determine_import_from_candidates(
     candidates: ParsedJavascriptDependencyCandidate,
     package_candidate_map: NodePackageCandidateMap,
+    file_extensions: tuple[str],
```

Comment:
To be able to reuse these functions, I've extended the function with an additional argument so that it can be used both for JS and TS rules.